code/modules/clothing/modular_armor/attachments/modules.dm

```diff
@@ -250,13 +250,28 @@
 
 /obj/item/armor_module/module/chemsystem/on_attach(obj/item/attaching_to, mob/user)
 	. = ..()
-	parent.AddComponent(/datum/component/chem_booster)
+	var/datum/component/chem_booster/chemsystem = parent.AddComponent(/datum/component/chem_booster)
+	RegisterSignal(chemsystem, COMSIG_CHEMSYSTEM_TOGGLE, .proc/update_module_icon)
 
 /obj/item/armor_module/module/chemsystem/on_detach(obj/item/detaching_from, mob/user)
 	var/datum/component/chem_booster/chemsystem = parent.GetComponent(/datum/component/chem_booster)
+	UnregisterSignal(chemsystem, COMSIG_CHEMSYSTEM_TOGGLE)
 	chemsystem.RemoveComponent()
 	return ..()
 
+///Updates the module on the armor to glow or not
+/obj/item/armor_module/module/chemsystem/proc/update_module_icon(datum/source, toggle)
+	SIGNAL_HANDLER
+	if(toggle)
+		icon_state = "mod_chemsystem_active"
+		item_state = "mod_chemsystem_active_a"
+		update_icon()
+		parent.update_icon()
+		return
+	icon_state = "mod_chemsystem"
+	item_state = "mod_chemsystem_a"
```

Comment:
this should still probs be in update_icon_state() and update_item_state(). Then this var can just set a var and call update_icon()